### PR TITLE
Added FileManager and APIClient classes, tests with pytest

### PR DESCRIPTION
**Pull request by JagnaJ**

### CODE REVIEW

File: weekThree/__pycache__/__init__.cpython-310.pyc
Comment (DmyMi) [inline weekThree/__pycache__/__init__.cpython-310.pyc:1]: Having a .gitignore is a must when you are doing tests :) as this compiled code should not appear in git. It will be a great Git excercise, because you can delete these folders from Git in a way that it looks like they never existed without breaking the repo :) if you will have trouble figuring it out - ping me.

File: weekThree/api_client.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class APIClient:
+    def __init__(self, url):
+        self.url = url
+
+    def fetch_data(self):
+        try:
+            response = requests.get(self.url)
+            response.raise_for_status()
+            return response.text
+        except requests.RequestException as e:
```
Comment (DmyMi) [inline weekThree/api_client.py:12]: Рaving general RequestException catch block is not perfect, because there can be network errors, server errors, client errors, not found (404, which some times is not considered as an error). It is better to have atleast a few more specialized exception handlers.

File: weekThree/file_manager.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    def count_lines(self):
+        if not os.path.exists(self.file_path):
+            print("File not found!")
+            return 0
+        with open(self.file_path, 'r', encoding='utf-8') as file:
+            return len(file.readlines())
+
+    def write_data(self, data):
+        with open(self.file_path, 'w', encoding='utf-8') as file:
```
Comment (DmyMi) [inline weekThree/file_manager.py:15]: You can improve write handling in file operations (e.g., check that the file exists before accessing it, so you do not overwrite some "important" file).

File: weekThree/file_manager.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+class FileManager:
+    def __init__(self, file_path):
+        self.file_path = file_path
+
+    def count_lines(self):
+        if not os.path.exists(self.file_path):
+            print("File not found!")
+            return 0
+        with open(self.file_path, 'r', encoding='utf-8') as file:
```
Comment (DmyMi) [inline weekThree/file_manager.py:11]: You can improve read handling in file operations, e.g., if the file is malformed or not a text file.

File: weekThree/tests/test_file_manager.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import pytest
+from weekThree.file_manager import FileManager
+
+def test_count_lines(tmp_path):
```
Comment (DmyMi) [inline weekThree/tests/test_file_manager.py:4]: While tests cover basic functionality, expand testing to include more edge cases and ensure file operations in tests are isolated (using temporary directories or pytest fixtures) to prevent side effects (by generating a temporary folders/files by Pytest).

File: weekThree/tests/test_api_client.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import pytest
+from weekThree.api_client import APIClient
+import requests
+from requests.exceptions import RequestException
+
+def test_fetch_data_success(monkeypatch):
```
Comment (DmyMi) [inline weekThree/tests/test_api_client.py:6]: Use fixtures and parameterized tests to test the APIClient functionality.

**Review by DmyMi — APPROVED** [review]
